### ISSUE TITLE
F5 starts debugging when inactive, but refreshes the page when active

### ISSUE THREAD

**Issue by rehmsen** [microsoft/vscode#187440]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.80
- OS Version: macOS 13.4.1

Steps to Reproduce:

I don't know how to reproduce this in VSCode, which does not have native browser refresh, or in vscode.dev, which does not have debugging, but this issue is occurring in Google's web version of VSCode that does support debugging, and would ideally be fixed upstream since you already distinguish isWeb and isPlatform and this will be broken when you or anybody does support debug on web. For us to reproduce it you start debugging with F5, and then press F5 again which then refreshes the page stopping the debugging etc.)

F5 in browsers is a fairly common shortcut for refresh. I think it should not be used to start debugging on web (`isWeb && !isStandalone()`). Instead, maybe we can use alt-F5 on web? That does not seem to be taken yet.

Please let me know if you would accept a contribution.

**Comment by boltex** [comment]
I also have some issues with the web version refreshing the page on F5 and ctrl+R, ... or ctrl+w that closes the browser tab.  (quite bad if hit ctrl + w accidentally)

Can vscode intercept those shortcuts bindings to block them even when no commands are bound to them?

**Comment by roblourens** [comment]
This is like https://github.com/microsoft/vscode/issues/181792. I wouldn't remove the existing keybinding on web, but I would take a PR that adds alt+F5 as an alternate keybinding.

**Comment by roblourens** [comment]
Actually, two things. It seems like that keybinding is actually claimed

<img width="576" alt="image" src="https://github.com/microsoft/vscode/assets/323878/6c7f95e0-62a7-4b10-82cb-f5f69a62de3e">

And in Edge and Chrome on Mac, F5 doesn't do anything. Maybe that is Windows only?

**Comment by rehmsen** [comment]
You are right, Mac does not seem to use F5 for refresh. Windows I don't have around for testing, but on my Linux F5 does refresh. This apparently unreliable source claims "all modern browsers use F5 for refreshing", which is clearly wrong on Mac, but may be true for the other OSes? https://www.computerhope.com/jargon/f/f5.htm#:~:text=In%20all%20modern%20Internet%20browsers,contents%20of%20the%20page%20again.

I think an alternative keyboard shortcut does not solve the problem, which is that depending on what state you are in, F5 does something very different, and while you are in a debug session it is destructive (F5 reloads the page and ends your debugging session).

I see two main directions:
1) VSCode overwrites F5, even on web. In that case I would like to make it also overwrite it, and possibly do nothing, in the cases not yet covered, e.g. when a debugger is running but not currently stopped at a breakpoint, to avoid the disruptive refresh which kills you debug session.
2) VSCode stops overwriting F5 on web because that is, at least on Linux and possibly on Windows, for refreshing the page. But then an alternative keyboard shortcut does not work - it needs to completely stop hogging F5 so that it can be used for refreshing always.

I think the current mixed state is very frustrating, at least for those on systems where this happens.

**Comment by boltex** [comment]
@rehmsen indeed, mixed behaviors are annoying. In my opinion, the 'refresh' is dangerous as it can be accidentally pressed. And since this is a 'web application' more than a 'webpage', i think the F5 should always be overridden - at worse to do 'nothing' 

A real, intentional 'refresh' can be still initiated by the user with the mouse press on the refresh button that all browsers do have.

**Comment by roblourens** [comment]
Guess I didn't follow up here- F5 does 'start' when `debugState == inactive` and 'continue' when `debugState == 'stopped'`. It would probably work to add a dummy command that runs when debugState is not inactive or stopped, just to fill the gap and block the keybinding from the browser.
